### ISSUE TITLE
Escaping special characters

### ISSUE THREAD

**Issue by EricKotato** [adifyr/obsidian-chat-view#22]
Is there a way to escape the vertical bar character `|`, so it can be shown in message?

Addressing escaping "less than" sign `<` in html tags would be also good, but it can be escaped by enclosing it in accents (\`<\`) while vertical bar can't be escaped at all.

**Comment by adifyr** [comment]
Hi @EricKotato. Thanks for the feedback. **Chat View V2** is currently in the works. I am thinking of replacing `|` with `~` which is a rarer character and overall simplifying the text interface.

This may be a slightly long term issue, will post updates here as the development cycle progresses.

**Comment by EricKotato** [comment]
Changing the character will break existing dialogs. With few dialogs it could be fine, but with many dialogs it would be certainly painful.

Using `|` itself as a part separator is fine, but the way to escape a character would be great, and I don't think that simply changing the character would help since I saw people using `~` as postfix very frequently.

**Comment by adifyr** [comment]
V2 will incorporate more changes than simply a character change. But, thanks for the feedback on which character is used frequently.

I'll see if I can give an escape sequence for the character in the meantime. Perhaps something like `\|` can be parsed as `|`. I'll get back to you on the same.

**Comment by adifyr** [comment]
Hi @EricKotato. I've published a fix to this issue with the release of Version 1.5.1 of Obsidian Chat View. You will now be able to safely escape `|` pipe characters by adding a `\` backslash in front of it.

Kindly confirm if this has been resolved on your end so that I may close the issue.

**Comment by EricKotato** [comment]
Looks like something broke.

This code:
````
```chat
< test message 1
> test message 2
```
````

Now results in this:
![](https://user-images.githubusercontent.com/2903496/211229505-775b7b1e-38eb-4f26-b2aa-929fe314506a.png)

Even this code:
````
```chat
< | test message 1
> | test message 2
```
````

Is now this:
![](https://user-images.githubusercontent.com/2903496/211229617-17fa2de8-8893-4215-aaa0-6c770a60bacf.png)

Even if I fill all fields:
````
```chat
< A | test message 1 | time
> B | test message 2 | time
```
````

It's shown like this:
![](https://user-images.githubusercontent.com/2903496/211229746-0d07fcfa-5649-44e0-bb88-a3445bf4a845.png)

**Comment by EricKotato** [comment]
Escaping, on the other hand, works as expected now:
![](https://user-images.githubusercontent.com/2903496/211229820-9458df7c-21a1-4d25-b51f-348fc7c7dbd8.png)

Code:
````
```chat
< A | test \| message 1 | time
> B | test \| message 2
```
````

**Comment by adifyr** [comment]
@EricKotato Thanks for letting me know. I'll look into it.

**Comment by adifyr** [comment]
I've just issued a fix in Release of Version 1.5.2. Can you check whether the bug has been fixed on your end? I'll close the issue subsequently.

**Comment by EricKotato** [comment]
Yes, works as expected now. Thanks!
